### PR TITLE
[DocumentIntelligence] Default authentication to Token Credential

### LINKED ISSUE #40054: [DocumentIntelligence] AAD not supported in 2023-10-31-preview
Error message:
```
Azure.RequestFailedException : ApiId document-intelligence-api-2023-10-31-preview OperationId GetInfo not supported for CheckAccess.
Status: 400 (Bad Request)
ErrorCode: API operation not supported for token authentication
```

Once AAD is supported, we need to re-enable tests:
- DocumentIntelligenceAdministrationClientCanAuthenticateWithTokenCredential
- DocumentIntelligenceClientCanAuthenticateWithTokenCredential

### PR DESCRIPTION
**Pull request by kinelski**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/40054.

### CODE REVIEW

**Comment by kinelski** [comment]
Just noticed that I haven't touched samples. Do we want to update samples to stop using key authentication as well?

**Comment by jsquire** [comment]
> Just noticed that I haven't touched samples. Do we want to update samples to stop using key authentication as well?

Outside of the "Hello World" and basic README samples, yes.   We should be demonstrating using identity-based auth.

**Comment by kinelski** [comment]
Updated samples to use `DefaultAzureCredential`.

File: sdk/documentintelligence/Azure.AI.DocumentIntelligence/tests/samples/SampleSnippets.cs [hunk 1]
```diff
@@ -18,25 +18,25 @@ public void CreateDocumentIntelligenceClient()
             #region Snippet:CreateDocumentIntelligenceClient
 #if SNIPPET
             string endpoint = "<endpoint>";
-            string apiKey = "<apiKey>";
 #else
             string endpoint = TestEnvironment.Endpoint;
-            string apiKey = TestEnvironment.ApiKey;
 #endif
-            var client = new DocumentIntelligenceClient(new Uri(endpoint), new AzureKeyCredential(apiKey));
+            var client = new DocumentIntelligenceClient(new Uri(endpoint), new DefaultAzureCredential());
```
Comment (jsquire) [inline sdk/documentintelligence/Azure.AI.DocumentIntelligence/tests/samples/SampleSnippets.cs:24]: Shouldn't all of these use `TestEnvironment.Credential` when running and show DAC injected into samples?   Pretty sure DAC is going to fail under federated auth in the pipelines.

Comment (kinelski) [reply sdk/documentintelligence/Azure.AI.DocumentIntelligence/tests/samples/SampleSnippets.cs:24]: That's a good point 🤦 Let me update it across the board.

**Comment by kinelski** [comment]
Comments addressed. Should be good now.

**Review by jsquire — APPROVED** [review]

**Pull request merged**